### PR TITLE
Implement system contract for `UPDATE_TOKEN`

### LINKED ISSUE #8593: UPDATE_TOKEN (TokenUpdatePrecompile)

### PR DESCRIPTION
**Pull request by dikel**
**Description**:
Implementation for HTS `updateTokenInfo`
1. Created `UpdateTranslator`
2. Created `UpdateDecoder`
3. Created different `wrapper` files in utils for more convenient usage of tuples
4. Added `XTest` cases
5. Written unit tests

**Related issue(s)**:

Fixes #8593 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-app/src/xtest/java/contract/UpdatesXTest.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                Account.newBuilder()
+                        .accountId(OWNER_ID)
+                        .alias(SENDER_ADDRESS)
+                        .key(SENDER_CONTRACT_ID_KEY)
+                        .smartContract(true)
+                        .build());
+        return accounts;
+    }
+
+    private static Address asAddress(String address) {
```
Comment (agadzhalov) [inline hedera-node/hedera-app/src/xtest/java/contract/UpdatesXTest.java:247]: Can we move `asAddress` method to a util file specially for X tests, something like `XUtils`? cc: @tinker-michaelj

Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/xtest/java/contract/UpdatesXTest.java:247]: Seems reasonable to me!

**Review by tinker-michaelj — APPROVED** [review]
LGTM, thanks @dikel !

File: hedera-node/hedera-app/src/xtest/java/contract/UpdatesXTest.java (hunk 1 above)
Comment (dikel) [reply hedera-node/hedera-app/src/xtest/java/contract/UpdatesXTest.java:247]: @agadzhalov created a separate issue (#9179) for this

**Pull request merged**